src/main/java/com/mraof/minestuck/client/gui/MinestuckScreen.java

```diff
@@ -47,10 +44,10 @@ public void drawGrist(PoseStack poseStack, int xOffset, int yOffset, int xcor, i
 		//Show the name of the grist instead of the count if displaying a tooltip
 		boolean showName = false;
 		GristType tooltipType = null;
-		GristSet clientGrist = ClientPlayerData.getClientGrist();
-		int clientRung = ClientPlayerData.getRung();
-		long cacheLimit = Echeladder.getGristCapacity(clientRung);
-
+		ClientPlayerData.ClientCache cache = ClientPlayerData.getGristCache(ClientEditHandler.isActive() ? ClientPlayerData.CacheSource.EDITMODE : ClientPlayerData.CacheSource.PLAYER);
+		GristSet clientGrist = cache.set();
+		long cacheLimit = cache.limit();
```

Comment:
Perhaps we should put the GristTorrent icon next to the number if its in the process of downloading grist from the gutter

Reply:
For this, I think we would need to sync which types of grist exist in the gutter, and have the torret icon display when the type in question exists in the gutter, as well as the cache is under capacity. This would then display for all grist types that *could* be sent from the gutter. We could of course instead consider displaying something when it actually gets sent.
Regardless, I think that it's outside the scope of this PR and could be looked at by someone after this.